### ISSUE TITLE
Drop Heroku support

### ISSUE THREAD

**Issue by jbrooksuk** [cachethq/Cachet#557]
Heroku are changing their pricing structure (from what we've seen), the "free" tier is no longer useful for Cachet anymore. On top of that, I've had terrible customer support - i.e. no replies - and supporting the Heroku config via Laravel 5's environment configurations is near impossible until 5.1.

- [x] Remove the "Deploy to Heroku" button.
- [x] Remove the `heroku` configuration.
- [x] Remove any mention of the "Deploy to Heroku" support.

We will not be supporting Heroku any longer.

CC @cachethq/owners.

**Comment by brianjking** [comment]
@jbrooksuk Bummer. Do you have suggestions on where we can deploy this easily/free elsewhere? I have a CentOS server and an AWS Instance (Micro) using ServerPilot. However, I really enjoyed Heroku for having it completely disconnected from any other infrastructure that we use for hosting the sites that I display the status of using Cachet.

Thanks...

**Comment by brianjking** [comment]
@jbrooksuk -- Perhaps viaduct.io? https://viaduct.io/features

**Comment by jbrooksuk** [comment]
Some people have suggested OpenShift; https://www.openshift.com/products/pricing/plan-comparison

I myself will be moving the demo and Cachet's own status page to a $5/pm droplet.

**Comment by StartledPhoenix** [comment]
I have an openshift deployment diy at https://github.com/StartledPhoenix/openshift-diy-cachet if you want to test.

**Comment by jbrooksuk** [comment]
Thanks @StartledPhoenix I've mentioned your project in our recent blog post https://blog.cachethq.io/dropping-native-heroku-support/

**Comment by sammcj** [comment]
Never been a fan of Heroku so Openshift has my vote for this!

**Comment by StartledPhoenix** [comment]
@jbrooksuk Almost done fixing some things, afterwards, it should allow me to give you a one-click button to install into openshift online. I've got some spare time for the next few days, so I can probably work all the kinks out by the end of this week.

A little question - composer.json has APCU listed, however OpenShift uses APC, can you add the check in for APC please - thanks.

**Comment by jbrooksuk** [comment]
@StartledPhoenix I don't think we can tell `composer.json` to use `ext-apcu` **or** `ext-apc`, @GrahamCampbell any advice?

**Comment by StartledPhoenix** [comment]
ext-apc provides both https://github.com/composer/composer/issues/3865

**Comment by jbrooksuk** [comment]
Make a PR :)

By the way @StartledPhoenix you'll want to be supporting the `L5` branch, which will become `master` soon.

**Comment by StartledPhoenix** [comment]
Ill get on L5 after master is confirmed to be working :)

**Comment by wuputah** [comment]
Hi James- we'd certainly be sad to see you purposely remove support for Heroku. If you have issues or feedback or Heroku, I'd encourage you to [open a support ticket](https://help.heroku.com/tickets/new) and we can route to the right teams/personnel. I can also try to direct you to the right people to @ mention on Github if you need help with something.

**Comment by jbrooksuk** [comment]
@wuputah thank you, but I feel it's a little too late now. Plus, what use is a status page that is only available for 12 hours of the day (when on the free plan). The free plan is what encouraged Cachet users to deploy to Heroku, without it, it seems silly supporting Heroku out of the box.

**Comment by wuputah** [comment]
The pricing beta feedback is @pvh's wheelhouse — it quite far from being final. We think the pricing experiment is a reasonable target for the service we provide (i.e., in this case, a 24x7 production status site), but we're open to feedback.

If you have technical concerns/feedback, we'd love to hear it as well! We always want it to be as easy as possible to run apps on Heroku—so you shouldn't have to go out of your way to support Heroku. So, we want to know about any pain points.

**Comment by pvh** [comment]
Hello! I'm sorry you've had a negative experience with Heroku. I can understand why you would draw the conclusions you have with the data you've got, but I hope I can offer a little counterbalancing perspective. 

We have presented a hypothesis for how to structure our free tier based on our reasonable but non-comprehensive understanding of the myriad ways people consume the platform. That said, the specific details of the free offering are very much up in the air. We love projects like yours and would hope that for people with light-duty applications a free deployment would work just fine. 

Should the deployment became heavily trafficked or frequently used enough to require 24x7 availability, we'd hope people would upgrade to a paid solution. After all, we pay (mostly) when the dyno is running. It's our ability to put applications to sleep during idle periods that lets us offer mostly a unlimited number of applications for free. 

I'm certainly open to discussing this further, though I suspect this Issue is not the most ideal venue. I'd be quite happy to hear more about how we could make the free tier appealing to you. It's definitely not our intent to drive projects like yours away. 

You can shoot me an email at pvh@heroku.com with your thoughts, but if you're past that point I'll just say that I'm very sad to see you go and I hope someday we'll earn your trust again.

**Comment by jbrooksuk** [comment]
Thanks @pvh and @wuputah, I appreciate you getting back to me. I completely understand *and* appreciate that you need to charge people as you can't continue to run a service without paying customers. I believe that it's because of your paying customers that you can even do the free tier in its existing form - please don't think that I don't realise how important it is for you as a company.

The problem I have with Heroku isn't your pricing restructure (I kind of get it), instead it's built up over the last few months of developing Cachet. Here is a list of the tweets where I've mentioned @heroku:

- https://twitter.com/jbrooksuk/status/562678418425516035
- https://twitter.com/jbrooksuk/status/546610493973344256
- https://twitter.com/jbrooksuk/status/546366756294762498
- https://twitter.com/jbrooksuk/status/538095816582639616
- https://twitter.com/jbrooksuk/status/537369563755986945

Considering that I even went as far as to write a [blog post](https://twitter.com/jbrooksuk/status/535706944099844096) on how to implement the Heroku Button because of the little documentation, I figured we'd get some kind of support? I know that I, myself do not pay for Heroku, but even if some of our users do, we've helped you out.

Apart from one or two people (whom I believe aren't Heroku employees) reaching out to me, I've had zero replies from yourselves.

As an aside, according to our anonymous (and optional) tracking we've seen 400 Heroku deployments:

![image](https://cloud.githubusercontent.com/assets/246103/6981299/bc0486d4-d9f6-11e4-8f23-bc9e57df5c7c.png)

If you'd like to speak further about this, then you can email me james@cachethq.io - I wanted to make my reasons public for dropping Heroku.

**Comment by wfjsw** [comment]
can you check out dotcloud.org plz? thanks.

**Comment by ApfelUser** [comment]
@wfjsw nice. Spam has arrived at GitHub. Nice. Awesome. Your service doesn't even offer a free plan -.-

**Comment by wfjsw** [comment]
@ApfelUser All right maybe I make something wrong...

**Comment by jbrooksuk** [comment]
@wfjsw it's not the fact that Heroku doesn't have a free plan that I won't support it...

**Comment by jbrooksuk** [comment]
I'll be dropping support for Heroku on the `L5` branch, it'll be left for prosperity.

**Comment by jbrooksuk** [comment]
Closed in https://github.com/cachethq/Cachet/pull/565.

**Comment by kfrankperson** [comment]
:+1: Great news !!!!

**Comment by wuputah** [comment]
@jbrooksuk I am the manager of the support team, and we also take support very seriously. I'm very sorry to hear that this impacted your decision to drop Heroku support in any way.  Alas, mentioning `@heroku` on Twitter (or Github) isn't something we try to derive support requests from. I'd instead invite you to open a support ticket anytime. 

Our marketing team does try to monitor the Twitter stream and tries to send things our way when it looks like technical outreach is needed, but I don't believe we monitor mentions, so our team never saw your tweets. I think if you had put `@heroku` at the beginning of your tweet rather than a mention, we would have been more likely to see it.

So, in short, the reason why you didn't hear from us is because we never heard you. The first time we heard of your complaints was when someone internally picked up the blog post, which led me here.

**Comment by jbrooksuk** [comment]
@wuputah I feel like I'm being blamed here for the reason that you didn't see my messages? It seems you're only hearing me now because I've been negative to Heroku...

**Comment by wuputah** [comment]
I'm only trying to explain what happened — or, more accurately, what _didn't_ happen — and, therefore, why we failed you in this case. I'm not trying to make excuses or place blame, but I am offering my apologies, and inviting you to contact us whenever you need help in a way which we will definitely respond.

We should definitely do a better job watching and responding to mentions on Twitter. That's definitely a failure by our organization to acknowledge Twitter as a serious channel for corporate communication, rather than just a marketing mechanism.

**Comment by troy** [comment]
As an outsider, I don't think it's reasonable to both (a) use Twitter, and also (b) expect a reply. No good answer fits in 140 characters, which means any response is either going to:

1. Be incomplete, often to the point of being useless or frustrating
2. Yield a reply like "What's your email address?"
3. Not get any reply, often because whoever sees the tweet realizes 1 above (and has seen 2 go nowhere hundreds of times)
4. Not get a reply, because out of however many hundreds of mentions this account receives every day, only a small portion want a reply

There's no right or wrong here, but I don't think it's fair to penalize an individual or a company for not responding to tweets. If you expect a reply, spend 2 minutes to use a contact method that fits the situation (and allows for a clear explanation of the question and the answer).

Heroku codified that in [how to reach us](https://www.heroku.com/policy/support), but even if they hadn't, among all of the ways to ask a question of any company, Twitter would be near the bottom of my list.

**Comment by troy** [comment]
An analogy came to mind. this seems a little like giving a one-star Yelp rating based on going to the restaurant and discovering that it closed 5 minutes before the closing time. The end result sucks - diner is not happy, restaurant loses business - and neither party would want it. However, it's not a representation of the dining (support) experience, and drawing conclusions based on it doesn't serve either party's interests. It's a one-line note in a larger review: "Heads up, they may close early" ("I wish Heroku provided support on Twitter").

**Comment by jbrooksuk** [comment]
Ok @troy, let me list my reasons for my decision.

1. Support has been terrible. If you're using Twitter as a way of communicating with people, make sure you actively check it. It's like the go to medium for a lot of companies. I've had support from many businesses through Twitter, your whole "Heroku aside, Tw is awful for support because only bad answers are 140c" [tweet](https://twitter.com/troyd/status/585111278420684800) is BS in my mind as I said, I've had great support in 140 characters. With that said, there are some occasions where you do need a customers email to provide detailed support.
2. Support was still terrible. I wasn't asking for PHP help, I was asking for support with their own Heroku Button. Nadda.
3. After my decision is made, I'm essentially told that it's my fault for the lack of support. No. That's not good enough. I tweeted several times (at different intervals) and had nothing. @wuputah has apologised for their lack of support and response and I have since accepted his apology.
4. We've helped make 400 installations to Heroku. They may not be unique customers, but that's a lot. A lot of people *could* have made an upgrade to a paid plan - for which we've helped Heroku out there.
5. Yes, I don't pay for Heroku, **but** (and here's the kicker Troy) I was going to for my future plans with Cachet. As it grows and features are added I'll have needed more dynos. **None** of this is about me not wanting to pay Heroku - I've [continuously said and supported the notion](https://news.ycombinator.com/item?id=9296310) that companies have to make money. On top of that, reducing the activity time for a dyno to half a day is of absolutely zero use to a status page which should be accessible at **all** times. You know, so people can check it.

**Comment by jfeulner** [comment]
I will have to agree with @jbrooksuk on this comment stream in regards to information. Why is this thread turning into a discussion of _HOW_ Twitter seems to be a bad medium when the standard is being set to companies around the globe by answering tweets/facebook and the various Social media platforms as a way to communicate to customers? In fact most of the responses to tweets that users send or asking for information point to a form or webpage to submit a full request to support. If the owner of this repo decides to drop support, he doesn't need to be told his request wasn't good enough.

The tactic or the method of how @jbrooksuk expressed his feelings are valid and honestly to attack the core of his argument by saying his tweet was not good enough or the opinions on how the service can be is a Strawman argument at best.

**Comment by troy** [comment]
Let's leave my opinion about whether Twitter is good or bad for support out of it. The point remains: Heroku could have the most abysmal or incredible support around, but what's been tested so far is simply whether they respond to tweets.

There's a great argument that they should respond to tweets (and maybe GitHub mentions, or StackOverflow mentions, or Discourse posts, or..), it's just a totally different one than having bad support.

All 5 items of @jbrooksuk's issues might well be completely true -- perhaps Heroku would have completely ignored him despite 400 installs -- but what's been tested so far is how well they respond to tweets. If responding to support on Twitter is absolutely a must-have, then that alone is a great reason to drop support. It's not a reason to conclude that Heroku has bad support, though, since @pvh noted "our team never saw your tweets."

**Comment by jbrooksuk** [comment]
> Let's leave my opinion about whether Twitter is good or bad for support out of it.

Let's not. Considering:

> As an outsider, I don't think it's reasonable to both (a) use Twitter, and also (b) expect a reply.

@troy I don't know what you're trying to achieve from this. I have several reasons (all above, please re-read) for dropping Heroku support. Do you want Heroku support in Cachet? Do you even use Cachet? Am I being stupid?

What really, really annoys me about the Twitter support (or lack thereof) isn't the lack of support from me, but that they have actively answered *other* people as well, so it's not like they *never* check their tweets.

And, for good measure, let me reiterate one of the core reasons for this:

> 5. Yes, I don't pay for Heroku, but (and here's the kicker Troy) I was going to for my future plans with Cachet. As it grows and features are added I'll have needed more dynos. None of this is about me not wanting to pay Heroku - I've continuously said and supported the notion that companies have to make money. On top of that, reducing the activity time for a dyno to half a day is of absolutely zero use to a status page which should be accessible at all times. You know, so people can check it.

**Comment by troy** [comment]
We've gone away from a useful discussion here, so I'm bowing out. Good luck and happy coding :)

**Comment by jbrooksuk** [comment]
@troy seriously, if you have anything else to say email me james@cachethq.io I'm interested in genuine reasons as to why I should continue to support Heroku.

**Comment by kfrankperson** [comment]
@jbrooksuk Just lock comments before it turns into a flame war.

**Comment by jbrooksuk** [comment]
@kfrankperson I'm leaving them open for the time being :)

**Comment by OliPicard** [comment]
Wow this has been quite a sad time :-1: , I was at a conference recently and spoke to some developers about the new pricing plan, They explained that the new pricing scheme would come into force in June and that if your app is used for 6 hours a day you don't have to pay. As a status system normally issues are resolved in a matter of a couple of hours making the Heroku free tier viable. I understand your frustration with being unable to communicate with Heroku on social media platforms, I've taken a quick look at deploying the app on alternative platforms, OpenShift fork seems like a good idea however the app is many commits behind the upstream. 

@jbrooksuk I'd recommend supporting One Click on OpenShift instead via the Github Page natively so that quick deployment is ready out of the box directly from you guys, It wouldn't take too long to setup and support. Alternatively providing a one click app on DigitalOcean might also be a way forward. One of the main reasons I decided to use your service was due to the easy of deployment, if it comes to it I will manually deploy this myself but was hoping not to so.

Hopefully something can be done to bridge the gap.
